Homematic: Optimized parameters and initialization

gerrieg opened the pull request.

gerrieg commented:
The scheduler.execute in the bridge and thing initialization does not fix the warning 'Initializing handler for thing ... takes more than 5000ms.'
How can i avoid this message? Initializing a thing could take longer than 5000ms.

kaikreuzer commented:
> How can i avoid this message? Initializing a thing could take longer than 5000ms.

The `initialize()` method has to return quickly - if this takes longer then 5s, you will see the warning.
But as I can see in your code, you seem to do the right thing: Schedule the long-running parts in a job that can run asynchronous then. Setting the ThingStatus to ONLINE/OFFLINE more than 5 seconds later is fully ok.

So can you confirm that your initialize() returns quickly? If this is the case and you nonetheless see the warning, we will have to analyse more closely why this happens.

kaikreuzer commented:
There seems to be some problem in the framework, see https://github.com/eclipse/smarthome/issues/2986

gerrieg commented:
It seems to be my fault. I built the homematic jar, copied the file into build 779. But it does not use the new jar file but the old one.

I replaced the homematic jar file in the kar file and in the folder ```userdata/kar/openhab-addons-2.1.0-SNAPSHOT/org/openhab/binding/org.openhab.binding.homematic/2.1.0-SNAPSHOT/org.openhab.binding.homematic-2.1.0-SNAPSHOT.jar``` but it still uses the old jar!?
I verified the jar with a java decompiler, it's the new one. I also deleted the ```userdata/cache/org.eclipse.osgi``` folder.

How can i install a new jar?  Is there a cache somewhere?

kaikreuzer commented:
Could you rebase this branch on the latest master, so that Github diff does not show the license header changes anymore?

gerrieg commented:
rebase done

kaikreuzer commented on addons/binding/org.openhab.binding.zoneminder/README.md:
```diff
@@ -1,176 +0,0 @@
-# Zoneminder Binding
```
not a good idea to delete readmes from other bindings :-)

gerrieg commented:
Ups :-) sorry, i had a pull conflict and forgot to revert the commit
